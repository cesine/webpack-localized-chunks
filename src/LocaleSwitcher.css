.locale-switcher {
  display: block;
}

a,a:visited {
  color: white;
}

a {
  margin-right: 10px;
}