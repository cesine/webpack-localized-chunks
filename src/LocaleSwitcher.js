import React from 'react';
import './LocaleSwitcher.css';

const LOCALES = ['en', 'fr'];
const routeChange = new CustomEvent('routeChange');

function changeUrl (locale) {
  window.history.pushState({
    locale,
  }, locale, `?locale=${locale}`);

  window.dispatchEvent(routeChange);
}

function LocaleSwitcher() {
  return (
    <p className="locale-switcher">
      {LOCALES.map((code, index) => (
        <a href={`?locale=${code}`} key={index} onClick={(e) => {e.preventDefault(); changeUrl(code) }}>{code}</a>
      ))}
      <a href="?">default</a>
    </p>
  );
}

export default LocaleSwitcher;
