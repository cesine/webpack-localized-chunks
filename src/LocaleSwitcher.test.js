import React from 'react';
import ReactDOM from 'react-dom';
import LocaleSwitcher from './LocaleSwitcher';

describe('LocaleSwitcher', () => {
  it('should use client side navigation', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LocaleSwitcher clientSide={true} />, div);
    expect(div).toMatchSnapshot();
  });

  it('should page reload navigation', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LocaleSwitcher clientSide={false} />, div);
    expect(div).toMatchSnapshot();
  });
});