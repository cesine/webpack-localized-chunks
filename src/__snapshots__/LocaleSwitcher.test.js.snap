// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleSwitcher should page reload navigation 1`] = `
<div>
  <p
    class="locale-switcher"
  >
    <a
      href="?clientSide=false&locale=en"
    >
      en
    </a>
    <a
      href="?clientSide=false&locale=fr"
    >
      fr
    </a>
    <a
      href="?"
    >
      default
    </a>
  </p>
</div>
`;

exports[`LocaleSwitcher should use client side navigation 1`] = `
<div>
  <p
    class="locale-switcher"
  >
    <a
      href="?clientSide=true&locale=en"
    >
      en
    </a>
    <a
      href="?clientSide=true&locale=fr"
    >
      fr
    </a>
    <a
      href="?"
    >
      default
    </a>
  </p>
</div>
`;
